File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution(object):
    def subarraySum(self, nums, k):
        d={0:1}
        sum=0
        c=0
        for i in nums:
            sum+=i
            if sum-k in d:
                c+=d[sum-k]
            d[sum]=d.get(sum,0)+1
        return c 

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
       HashMap<Integer,Integer> freq=new HashMap<>();
       freq.put(0,1);
       int sum=0;
       int c=0;
       for(int x:nums){
         sum+=x;
         if(freq.containsKey(sum-k)){
            c+=freq.get(sum-k);

         }
         freq.put(sum,freq.getOrDefault(sum,0)+1);
       }
    return c; 
    }
}